File: ViewPager/app/src/main/java/com/example/viewpager/MainActivity.java
package com.example.viewpager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        init();
    }

    private void init() {
        ViewPager viewPager=findViewById(R.id.viewpager);
        FragmentManager messageFragmentManager = getSupportFragmentManager();
        ViewPageAdapter viewPageAdapter = new ViewPageAdapter(this,viewPager, messageFragmentManager);
        viewPager.setAdapter(viewPageAdapter);
    }
}

File: custom_dialog/safeDialog/app/src/main/java/com/cxyzy/safedialog/MainActivity.kt
package com.cxyzy.safedialog

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textView.setOnClickListener {
            startActivity(
                Intent(
                    this@MainActivity,
                    SecondActivity::class.java
                )
            )
        }
    }
}
